cli/slsa-verifier/main_test.go

```diff
@@ -457,3 +444,179 @@ func Test_runVerify(t *testing.T) {
 		})
 	}
 }
+
+func TestContainerVerification(t *testing.T) {
+	// Override cosign image verification function for local image testing.
+	container.RunCosignImageVerification = func(ctx context.Context,
+		imageReference string, co *cosign.CheckOpts) ([]oci.Signature, bool, error) {
+		return cosign.VerifyLocalImageAttestations(ctx, imageReference, co)
+	}
+
+	// TODO: Is there a more uniform way of handling getting image digest for both
+	// remote and local images?
+	container.GetImageDigest = func(imageReference string) (string, error) {
+		// This is copied from cosign's VerifyLocalImageAttestation code:
+		// https://github.com/sigstore/cosign/blob/fdceee4825dc5d56b130f3f431aab93137359e79/pkg/cosign/verify.go#L654
+		se, err := layout.SignedImageIndex(imageReference)
+		if err != nil {
+			return "", err
+		}
+
+		var h v1.Hash
+		// Verify either an image index or image.
+		ii, err := se.SignedImageIndex(v1.Hash{})
+		if err != nil {
+			return "", err
+		}
+		i, err := se.SignedImage(v1.Hash{})
+		if err != nil {
+			return "", err
+		}
+		switch {
+		case ii != nil:
+			h, err = ii.Digest()
+			if err != nil {
+				return "", err
+			}
+		case i != nil:
+			h, err = i.Digest()
+			if err != nil {
+				return "", err
+			}
+		default:
+			return "", errors.New("must verify either an image index or image")
+		}
+		return strings.TrimPrefix(h.String(), "sha256:"), nil
+	}
+
+	// t.Parallel()
```

Comment:
add a comment why need to disable parallelism

Reply:
These can be made in parallel: i'm not worried about rate limiting, most will be using offline bundle verification